### DISCUSSION TITLE
Order By Random()

### DISCUSSION THREAD

**Discussion by liho00** [directus/directus#12333] [Archive]
I dont seem to find order by random options in the global queries & in option sort: {}
I think this quite useful when doing data analysis & selecting random sample from database.
Possible this is a feature request or we do have existing solution for it?

```
Select a random row with PostgreSQL:
SELECT column FROM table
ORDER BY RANDOM()
LIMIT 100
```

Reference link: https://www.petefreitag.com/item/466.cfm

**Comment by rijkvanzanten** [comment]
There's currently no way to order randomly in Directus. See https://github.com/directus/directus/discussions/3240 for the corresponding feature request 👍🏻

**Reply by liho00** [reply]
I got an alternative solution, but not sure does it workable or not, which is to create an api extension and then use the database instance to randomly select the data from database. Do you think this is a workaround way to achieve?

**Reply by rijkvanzanten** [reply]
Totally! Custom extensions can always be used as an escape hatch for nearly anything 🙂
